src/erldantic_json_schema.erl

```diff
@@ -0,0 +1,388 @@
+                                      ({map_field_type_assoc, KeyType, ValueType}) ->
+                                          {map_field_type_assoc,
+                                           type_replace_vars(TypeInfo, KeyType, NamedTypes),
+                                           type_replace_vars(TypeInfo, ValueType, NamedTypes)};
+                                      ({map_field_type_exact, KeyType, ValueType}) ->
+                                          {map_field_type_exact,
+                                           type_replace_vars(TypeInfo, KeyType, NamedTypes),
+                                           type_replace_vars(TypeInfo, ValueType, NamedTypes)}
+                                  end,
+                                  Fields)};
+        #ed_rec_ref{record_name = RecordName, field_types = RefFieldTypes} ->
+            case TypeInfo of
+                #{{record, RecordName} := #ed_rec{fields = Fields} = Rec} ->
+                    NewRec = Rec#ed_rec{fields = record_replace_vars(Fields, RefFieldTypes)},
+                    type_replace_vars(TypeInfo, NewRec, NamedTypes);
+                #{} ->
+                    erlang:error({missing_type, {record, RecordName}})
+            end;
+        #ed_remote_type{mfargs = {Module, TypeName, Args}} ->
+            case erldantic_module_types:get(Module) of
+                {ok, TypeInfo} ->
+                    TypeArity = length(Args),
+                    case TypeInfo of
+                        #{{type, TypeName, TypeArity} := Type} ->
+                            type_replace_vars(TypeInfo, Type, NamedTypes);
+                        #{} ->
+                            erlang:error({missing_type, TypeName})
+                    end;
+                {error, _} = Err ->
+                    erlang:error(Err)
+            end;
+        #ed_list{type = ListType} ->
+            #ed_list{type = type_replace_vars(TypeInfo, ListType, NamedTypes)}
+    end;
+type_replace_vars(_TypeInfo, #ed_rec{fields = Fields} = Rec, NamedTypes) ->
+    Rec#ed_rec{fields =
+                   lists:map(fun({Name, NType}) ->
+                                {Name, type_replace_vars(_TypeInfo, NType, NamedTypes)}
+                             end,
+                             Fields)};
+type_replace_vars(_TypeInfo, Type, _NamedTypes) ->
+    Type.
+
+arg_names(#ed_type_with_variables{vars = Args}) ->
+    Args;
+arg_names(_) ->
+    [].
+
+apply_args(TypeInfo, Type, TypeArgs) when is_list(TypeArgs) ->
+    ArgNames = arg_names(Type),
+    NamedTypes =
+        maps:from_list(
+            lists:zip(ArgNames, TypeArgs)),
+    type_replace_vars(TypeInfo, Type, NamedTypes).
+
+-spec map_fields_to_schema(erldantic:type_info(), [erldantic:map_field()]) ->
+                              {ok, map()} | {error, [erldantic:error()]}.
+map_fields_to_schema(TypeInfo, Fields) ->
+    case process_map_fields(TypeInfo, Fields, #{}, [], false) of
+        {ok, Properties, Required, HasAdditional} ->
```

Comment:
It seems to me that HasAdditional will always be false. When should it be set to true?